x/ibc/core/04-channel/client/utils/utils.go

```diff
@@ -232,6 +193,86 @@ func queryNextSequenceRecvABCI(clientCtx client.Context, portID, channelID strin
 	return types.NewQueryNextSequenceReceiveResponse(sequence, proofBz, proofHeight), nil
 }
 
+// QueryPacketCommitment returns a packet commitment.
+// If prove is true, it performs an ABCI store query in order to retrieve the merkle proof. Otherwise,
+// it uses the gRPC query client.
+func QueryPacketCommitment(
+	clientCtx client.Context, portID, channelID string,
+	sequence uint64, prove bool,
+) (*types.QueryPacketCommitmentResponse, error) {
+	if prove {
+		return queryPacketCommitmentABCI(clientCtx, portID, channelID, sequence)
+	}
+
+	queryClient := types.NewQueryClient(clientCtx)
+	req := &types.QueryPacketCommitmentRequest{
+		PortId:    portID,
+		ChannelId: channelID,
+		Sequence:  sequence,
+	}
+
+	return queryClient.PacketCommitment(context.Background(), req)
+}
+
+func queryPacketCommitmentABCI(
+	clientCtx client.Context, portID, channelID string, sequence uint64,
+) (*types.QueryPacketCommitmentResponse, error) {
+	key := host.KeyPacketCommitment(portID, channelID, sequence)
+
+	value, proofBz, proofHeight, err := ibcclient.QueryTendermintProof(clientCtx, key)
+	if err != nil {
+		return nil, err
+	}
+
+	// check if packet commitment exists
+	if len(value) == 0 {
+		return nil, sdkerrors.Wrapf(types.ErrPacketCommitmentNotFound, "portID (%s), channelID (%s), sequence (%d)", portID, channelID, sequence)
+	}
+
+	return types.NewQueryPacketCommitmentResponse(value, proofBz, proofHeight), nil
+}
+
+// QueryPacketReceipt returns data about a packet receipt.
+// If prove is true, it performs an ABCI store query in order to retrieve the merkle proof. Otherwise,
+// it uses the gRPC query client.
+func QueryPacketReceipt(
+	clientCtx client.Context, portID, channelID string,
+	sequence uint64, prove bool,
+) (*types.QueryPacketReceiptResponse, error) {
+	if prove {
+		return queryPacketReceiptABCI(clientCtx, portID, channelID, sequence)
+	}
+
+	queryClient := types.NewQueryClient(clientCtx)
+	req := &types.QueryPacketReceiptRequest{
+		PortId:    portID,
+		ChannelId: channelID,
+		Sequence:  sequence,
+	}
+
+	return queryClient.PacketReceipt(context.Background(), req)
+}
+
+func queryPacketReceiptABCI(
+	clientCtx client.Context, portID, channelID string, sequence uint64,
+) (*types.QueryPacketReceiptResponse, error) {
+	key := host.KeyPacketReceipt(portID, channelID, sequence)
+
+	value, proofBz, proofHeight, err := ibcclient.QueryTendermintProof(clientCtx, key)
+	if err != nil {
+		return nil, err
+	}
+
+	var receipt string
+	if value == nil {
+		receipt = ""
+	} else {
+		receipt = string(value)
+	}
```

Comment:
```suggestion
	if value != nil {
		receipt = string(value)
	}
```

Reply:
Receipt should always be empty strings so I don't understand why we even need this?